backend/composer/management/commands/ingest_anatomical_entities.py

```diff
@@ -1,64 +1,88 @@
 import csv
-
-from django.core.management.base import BaseCommand
-
-from composer.models import AnatomicalEntity
+import time
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.db.utils import IntegrityError
+from composer.models import AnatomicalEntity, Synonym
 
 URI = "o"
 NAME = "o_label"
 SYNONYM = "o_synonym"
+BULK_LIMIT = 100
 
 
 class Command(BaseCommand):
-    help = "Ingests Anatomical Entities CSV file(s)"
+    help = "Ingests Anatomical Entities CSV file(s), with robust error handling."
 
     def add_arguments(self, parser):
         parser.add_argument("csv_files", nargs="+", type=str)
+        parser.add_argument("--update_names", action='store_true',
+                            help="Update the name even if the entity already exists")
+        parser.add_argument("--show_complete_logs", action='store_true',
+                            help="Show detailed logs during processing")
+
+    def _process_anatomical_entity(self, name, ontology_uri, synonym, update_names, show_complete_logs, processed_uris,
+                                   synonym_accumulator):
+        try:
+            is_first_occurrence = ontology_uri not in processed_uris
```

Comment:
question: @afonsobspinto what is the use case for the is first occurence? What if the next occurence of the (same) uri has a different name? I don't see any error handling for that and I also don't see any logging. Imo this is either over engineering or non finiished functionality

Reply:
The name we want to keep is the **first** one that appears. 

If the next occurrence of the same uri has a different name we do **not** want to update the name.

See [this thread ](https://metacell.slack.com/archives/C0477SPSQQK/p1710368649332379?thread_ts=1710352121.100009&cid=C0477SPSQQK)